kernel/kernel-api/src/main/java/io/delta/kernel/internal/skipping/DataSkippingUtils.java

```diff
@@ -179,6 +179,23 @@ private static Optional<Predicate> constructDataSkippingFilter(
                     return e2Filter; // possibly none
                 }
 
+            // Match any file whose null count is less than the row count.
+            case "IS_NOT_NULL":
+                Expression child = getUnaryChild(dataFilters);
+                if (child instanceof Column) {
+                    Column childColumn = (Column) child;
+                    if (schemaHelper.isSkippingEligibleNullCountColumn((Column) child)) {
+                        return Optional.of(
+                            new Predicate(
+                                "<",
```

Comment:
I feel like we should have utilities to create expressions like `less(left, right)`, that makes it easy to understand and less code.

Reply:
Makes sense but I'd rather do this in a separate PR and clean up all existing usages